### PR TITLE
Add git-stree.

### PR DESCRIPTION
**Pull request by a1russell**

### CODE REVIEW

File: Library/Formula/git-stree.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  url "https://github.com/tdd/git-stree/archive/v0.4.0.tar.gz"
+  sha256 "c04d19c8eb1a566a8e7f373d447ab7e3480ee95f94802fa957afeb22c4bed0ef"
+
+  def install
+    bin.install("git-stree")
+    bash_completion.install("git-stree-completion.bash" => "git-stree")
+  end
+
+  test do
+    shell_output("git stree help 2>&1").include?("Usage:")
```
Comment (mikemcquaid) [inline Library/Formula/git-stree.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/git-stree.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class GitStree < Formula
+  homepage "https://github.com/tdd/git-stree"
+  head "https://github.com/tdd/git-stree.git"
+  url "https://github.com/tdd/git-stree/archive/v0.4.0.tar.gz"
+  sha256 "c04d19c8eb1a566a8e7f373d447ab7e3480ee95f94802fa957afeb22c4bed0ef"
+
+  def install
+    bin.install("git-stree")
```
Comment (mikemcquaid) [inline Library/Formula/git-stree.rb]: Don't need parens here (or indeed anywhere they except `shell_output` below)

File: Library/Formula/git-stree.rb (hunk 1 above)
Comment (a1russell) [inline Library/Formula/git-stree.rb]: No, I'd rather not have the test pull someone's git repo as a subtree just for a test.

Comment (mikemcquaid) [inline Library/Formula/git-stree.rb]: I'd rather not include this into Homebrew without a good test.

Comment (a1russell) [inline Library/Formula/git-stree.rb]: What should be the purpose of the test? Do you want to test the software, or that the software is installed? I don't understand why you think the test is bad.

Comment (mikemcquaid) [inline Library/Formula/git-stree.rb]: The purpose of the test is do something other than `--version` or `--help` which both are just outputting a string without exercising any other logic.

Comment (mikemcquaid) [inline Library/Formula/git-stree.rb]: It would be nice to create a new repo with `git init` and then use `git stree add` to pull in a subtree, thanks.

Comment (a1russell) [inline Library/Formula/git-stree.rb]: Done.

**Comment by mikemcquaid** [comment]
Seems to be some UTF-8 issues: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/23237/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_git_stree/

Maybe just drop the `_match` and use `system` which will check the return code. Thanks!

**Comment by a1russell** [comment]
Did as you suggested, @mikemcquaid.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `git-stree 0.4.0 (new formula)`.

**Pull request closed without merging**